profiling/src/config.rs

```diff
@@ -942,6 +960,18 @@ pub(crate) fn minit(module_number: libc::c_int) {
                     displayer: None,
                     env_config_fallback: None,
                 },
+                zai_config_entry {
+                    id: transmute::<ConfigId, u16>(ProfilingAllocationSamplingDistance),
+                    name: ProfilingAllocationSamplingDistance.env_var_name(),
+                    type_: ZAI_CONFIG_TYPE_CUSTOM,
+                    default_encoded_value: ZaiStr::literal(b"4194304\0"), // 4194304 = 4 MB
```

Comment:
Maybe leave the name of the extracted constant  in a comment here so if anyone changes it, they remember to change it here too.